tests/components/mqtt_eventstream/test_init.py

```diff
@@ -25,177 +23,170 @@ def mock_storage(hass_storage):
-        assert self.add_eventstream(pub_topic="bar")
-        self.hass.block_till_done()
-
-        # Reset the mock because it will have already gotten calls for the
-        # mqtt_eventstream state change on initialization, etc.
-        mock_pub.reset_mock()
+async def test_setup_succeeds(hass, mqtt_mock):
+    """Test the success of the setup."""
+    assert await add_eventstream(hass)
 
-        fire_time_changed(self.hass, dt_util.utcnow())
-        assert not mock_pub.called
 
-    def test_receiving_remote_event_fires_hass_event(self):
-        """Test the receiving of the remotely fired event."""
-        sub_topic = "foo"
-        assert self.add_eventstream(sub_topic=sub_topic)
-        self.hass.block_till_done()
+async def test_setup_with_pub(hass, mqtt_mock):
+    """Test the setup with subscription."""
+    # Should start off with no listeners for all events
+    assert hass.bus.async_listeners().get("*") is None
 
-        calls = []
+    assert await add_eventstream(hass, pub_topic="bar")
+    await hass.async_block_till_done()
 
-        @callback
-        def listener(_):
-            calls.append(1)
+    # Verify that the event handler has been added as a listener
+    assert hass.bus.async_listeners().get("*") == 1
 
-        self.hass.bus.listen_once("test_event", listener)
-        self.hass.block_till_done()
 
-        payload = json.dumps(
-            {"event_type": "test_event", "event_data": {}}, cls=JSONEncoder
-        )
-        fire_mqtt_message(self.hass, sub_topic, payload)
-        self.hass.block_till_done()
+async def test_subscribe(hass, mqtt_mock):
+    """Test the subscription."""
+    sub_topic = "foo"
+    assert await add_eventstream(hass, sub_topic=sub_topic)
+    await hass.async_block_till_done()
 
-        assert 1 == len(calls)
+    # Verify that the this entity was subscribed to the topic
+    mqtt_mock.async_subscribe.assert_called_with(sub_topic, ANY, 0, ANY)
 
-    @patch("homeassistant.components.mqtt.async_publish")
-    def test_ignored_event_doesnt_send_over_stream(self, mock_pub):
-        """Test the ignoring of sending events if defined."""
-        assert self.add_eventstream(pub_topic="bar", ignore_event=["state_changed"])
-        self.hass.block_till_done()
 
-        e_id = "entity.test_id"
-        event = {}
-        event["event_type"] = EVENT_STATE_CHANGED
-        new_state = {"state": "on", "entity_id": e_id, "attributes": {}}
-        event["event_data"] = {"new_state": new_state, "entity_id": e_id}
+async def test_state_changed_event_sends_message(hass, mqtt_mock):
+    """Test the sending of a new message if event changed."""
+    now = dt_util.as_utc(dt_util.now())
+    e_id = "fake.entity"
+    pub_topic = "bar"
+    with patch(
+        ("homeassistant.core.dt_util.utcnow"), return_value=now,
+    ):
+        # Add the eventstream component for publishing events
+        assert await add_eventstream(hass, pub_topic=pub_topic)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)